backend/src/main/java/com/benine/backend/http/CreatePresetHandler.java

```diff
@@ -60,11 +61,15 @@ public void handle(String s, Request request, HttpServletRequest req, HttpServle
         IPCamera ipcam = (IPCamera) camera;
         setPreset(ipcam, tagList);
         respondSuccess(request, res);
+      } else if (camera instanceof SimpleCamera) {
```

Comment:
I think the instanceof is a code smell. You might want to consider moving the important parts of the code within this statement to the SimpleCamera class maybe even create a presetCamera interface or something for a camera that supports presets.